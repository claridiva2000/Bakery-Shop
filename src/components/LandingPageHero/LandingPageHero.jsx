import { Link } from 'react-router-dom';
import Slider from 'react-animated-slider';
import content from './slides';
import CustomButton from '../custom-button/custom-button.component';
import './landingpagehero.styles.scss';

function LandingPageHero() {
  return (
    <div>
      <Slider>
        {content.map((item, index) => (
          <div
            key={index}
            className="slideBG"
            style={{
              background: `url('${item.image}') no-repeat center center`,
              backgroundSize: `cover`,
              outline: '200px solid rgba(0, 0, 0, 0.4)',
              outlineOffset: '-200px',
              overflow: 'hidden',
            }}
          >
            <div className="center">
              <h1>{item.title}</h1>
              <p>{item.description}</p>
              <Link to="/shop">
                <CustomButton>Order Now</CustomButton>
              </Link>
            </div>
          </div>
        ))}
      </Slider>
    </div>
  );
}

export default LandingPageHero;
