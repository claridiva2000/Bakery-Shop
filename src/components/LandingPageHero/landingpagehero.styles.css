@import url("https://fonts.googleapis.com/css2?family=Playfair+Display:ital,wght@1,400;1,500&display=swap");
@import url("https://fonts.googleapis.com/css?family=Nunito|Quicksand");
@import url("https://fonts.googleapis.com/css2?family=Playfair+Display:ital,wght@1,400;1,500&display=swap");
@import url("https://fonts.googleapis.com/css?family=Nunito|Quicksand");
*,
*::after,
*::before {
  margin: 0;
  padding: 0;
  -webkit-box-sizing: inherit;
          box-sizing: inherit;
}

body {
  font-weight: normal;
  font-family: "Quicksand", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-rendering: optimizeLegibility;
}

h1,
h2,
h3,
h4,
h5,
h6 {
  font-family: "Quicksand", sans-serif, sans-serif;
}

.slider {
  position: relative;
  width: 100%;
  height: 400px;
  overflow: hidden;
}

.slider a.previousButton, .slider a.nextButton {
  font-size: 22px;
  line-height: 0;
  display: block;
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transition: all .3s linear;
  transition: all .3s linear;
  z-index: 1;
  color: #333;
  padding: 10px;
  text-decoration: none;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
}

.slider a.previousButton:not(.disabled):hover, .slider a.nextButton:not(.disabled):hover {
  -webkit-transform: translateY(-50%) scale(1.25);
  transform: translateY(-50%) scale(1.25);
  cursor: pointer;
}

.slider a.previousButton {
  left: 20px;
}

.slider a.nextButton {
  right: 20px;
}

.slide {
  width: 100%;
  height: 100%;
  position: absolute;
  overflow: hidden;
}

.slide.hidden {
  visibility: hidden;
}

.slide.previous {
  left: -100%;
}

.slide.current {
  left: 0;
}

.slide.next {
  left: 100%;
}

.slide.animateIn,
.slide.animateOut {
  -webkit-transition: all 2s ease;
  transition: all 2s ease;
}

.slide.animateIn.previous,
.slide.animateIn.next {
  left: 0;
  visibility: visible;
}

.slide.animateOut.previous {
  left: 100%;
}

.slide.animateOut.next {
  left: -100%;
}

.center {
  z-index: 200;
  color: mintcream;
  text-align: center;
  position: relative;
  top: 28%;
  left: 21%;
  width: 60%;
}

.center h1 {
  margin-bottom: 40px;
}

.center p {
  margin-bottom: 40px;
  line-height: 19px;
}

polygon {
  fill: #a13d63;
  z-index: 100;
  -webkit-filter: drop-shadow(7 7 0.75rem #fafafa);
          filter: drop-shadow(7 7 0.75rem #fafafa);
}

.sliderbtn {
  background: #ffffff;
  padding: 10px;
  border: none;
  font-weight: 700;
  border-radius: 8px;
  -webkit-box-shadow: 0 4px 8px 0 rgba(7, 7, 7, 0.5);
          box-shadow: 0 4px 8px 0 rgba(7, 7, 7, 0.5);
}
/*# sourceMappingURL=landingpagehero.styles.css.map */