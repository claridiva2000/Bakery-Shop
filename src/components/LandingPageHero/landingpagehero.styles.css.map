{
    "version": 3,
    "mappings": "AEAA,OAAO,CAAC,mGAAI;AACZ,OAAO,CAAC,+DAAI;AADZ,OAAO,CAAC,mGAAI;AACZ,OAAO,CAAC,+DAAI;ADCZ,AAAA,CAAC;AACD,CAAC,AAAA,OAAO;AACR,CAAC,AAAA,QAAQ,CAAC;EACR,MAAM,EAAE,CAAC;EACT,OAAO,EAAE,CAAC;EACV,UAAU,EAAE,OAAO;CACpB;;AAED,AAAA,IAAI,CAAC;EACH,WAAW,EAAE,MAAM;EACnB,WAAW,ECFE,WAAW,EAAE,UAAU;EDGpC,sBAAsB,EAAE,WAAW;EACnC,uBAAuB,EAAE,SAAS;EAClC,cAAc,EAAE,kBAAkB;CACnC;;AAED,AAAA,EAAE;AACF,EAAE;AACF,EAAE;AACF,EAAE;AACF,EAAE;AACF,EAAE,CAAC;EACD,WAAW,ECdE,WAAW,EAAE,UAAU,EDcR,UAAU;CACvC;;ADtBD,AAAA,OAAO,CAAC;EACP,QAAQ,EAAE,QAAQ;EAClB,KAAK,EAAE,IAAI;EACX,MAAM,EAAE,KAAK;EACV,QAAQ,EAAE,MAGd;CAAC;;AAED,AAAA,OAAO,CAAC,CAAC,AAAA,eAAe,EAAE,OAAO,CAAC,CAAC,AAAA,WAAW,CAAC;EAC9C,SAAS,EAAE,IAAI;EACf,WAAW,EAAE,CAAC;EACd,OAAO,EAAE,KAAK;EACd,QAAQ,EAAE,QAAQ;EAClB,GAAG,EAAE,GAAG;EACR,iBAAiB,EAAE,gBAAgB;EAC3B,SAAS,EAAE,gBAAgB;EACnC,kBAAkB,EAAE,cAAc;EAClC,UAAU,EAAE,cAAc;EAC1B,OAAO,EAAE,CAAC;EACV,KAAK,EAAE,IAAI;EACX,OAAO,EAAE,IAAI;EACb,eAAe,EAAE,IAAI;EACrB,2BAA2B,EAAE,MAAM;EAC3B,mBAAmB,EAAE,MAAM;CAEnC;;AAED,AAAA,OAAO,CAAC,CAAC,AAAA,eAAe,AAAA,IAAK,CAAA,SAAS,CAAC,MAAM,EAAE,OAAO,CAAC,CAAC,AAAA,WAAW,AAAA,IAAK,CAAA,SAAS,CAAC,MAAM,CAAC;EACxF,iBAAiB,EAAE,gBAAgB,CAAC,WAAW;EACvC,SAAS,EAAE,gBAAgB,CAAC,WAAW;EAC/C,MAAM,EAAE,OAAO;CACf;;AAED,AAAA,OAAO,CAAC,CAAC,AAAA,eAAe,CAAC;EACxB,IAAI,EAAE,IAAI;CACV;;AAED,AAAA,OAAO,CAAC,CAAC,AAAA,WAAW,CAAC;EACpB,KAAK,EAAE,IAAI;CACX;;AAED,AAAA,MAAM,CAAC;EACN,KAAK,EAAE,IAAI;EACX,MAAM,EAAE,IAAI;EACZ,QAAQ,EAAE,QAAQ;EAClB,QAAQ,EAAE,MACX;CAAC;;AAED,AAAA,MAAM,AAAA,OAAO,CAAC;EACb,UAAU,EAAE,MAAM;CAClB;;AAED,AAAA,MAAM,AAAA,SAAS,CAAC;EACf,IAAI,EAAE,KAAK;CACX;;AAED,AAAA,MAAM,AAAA,QAAQ,CAAC;EACd,IAAI,EAAE,CAAC;CACP;;AAED,AAAA,MAAM,AAAA,KAAK,CAAC;EACX,IAAI,EAAE,IAAI;CACV;;AAED,AAAA,MAAM,AAAA,UAAU;AACf,MAAM,AAAA,WAAW,CAAC;EAClB,kBAAkB,EAAE,WAAW;EAC/B,UAAU,EAAE,WAAW;CACvB;;AAED,AAAA,MAAM,AAAA,UAAU,AAAA,SAAS;AACvB,MAAM,AAAA,UAAU,AAAA,KAAK,CAAC;EACvB,IAAI,EAAE,CAAC;EACP,UAAU,EAAE,OAAO;CACnB;;AAED,AAAA,MAAM,AAAA,WAAW,AAAA,SAAS,CAAC;EAC1B,IAAI,EAAE,IAAI;CACV;;AAED,AAAA,MAAM,AAAA,WAAW,AAAA,KAAK,CAAC;EACtB,IAAI,EAAE,KAAK;CACX;;AAED,AAAA,OAAO,CAAA;EACJ,OAAO,EAAE,GAAG;EACZ,KAAK,EAAE,SAAS;EAEhB,UAAU,EAAE,MAAM;EAClB,QAAQ,EAAE,QAAQ;EAClB,GAAG,EAAE,GAAG;EACR,IAAI,EAAE,GAAG;EACT,KAAK,EAAE,GAAG;CASZ;;AAjBD,AASG,OATI,CASJ,EAAE,CAAA;EACE,aAAa,EAAE,IAAI;CACtB;;AAXJ,AAYG,OAZI,CAYJ,CAAC,CAAA;EACG,aAAa,EAAE,IAAI;EACnB,WAAW,EAAE,IAAI;CACpB;;AAGJ,AAAA,OAAO,CAAA;EACH,IAAI,EEtGC,OAAO;EFuGZ,OAAO,EAAE,GAAG;EACZ,MAAM,EAAE,gCAA2C;CAEtD;;AAED,AAAA,UAAU,CAAA;EACN,UAAU,EEzGN,OAAO;EF0GX,OAAO,EAAE,IAAI;EACb,MAAM,EAAE,IAAI;EACZ,WAAW,EAAE,GAAG;EAChB,aAAa,EAAE,GAAG;EAClB,UAAU,EAAE,CAAC,CAAC,GAAG,CAAC,GAAG,CAAC,CAAC,CAAC,kBAAkB;CAE7C",
    "sources": [
        "landingpagehero.styles.scss",
        "../../scss/_base.scss",
        "../../scss/_variables.scss",
        "../../scss/_variables.scss"
    ],
    "names": [],
    "file": "landingpagehero.styles.css"
}