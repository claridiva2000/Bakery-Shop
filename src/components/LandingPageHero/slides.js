export default [
  {
    title: 'Vulputate Mollis Ultricies',
    description:
      'Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis vestibulum. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh.',
    button: 'Read More',
    image:
      'https://firebasestorage.googleapis.com/v0/b/bakery-dev-server.appspot.com/o/images%2Fbanana%20pudding.png?alt=media&token=319d993f-6ebf-435d-bbd0-ca2c5e96d130',
    user: 'Daniel',
    userProfile: 'https://s7.postimg.cc/abavelo3v/1_3x.png',
  },
  {
    title: 'Tortor Dapibus',
    description:
      'Nullam id dolor id nibh ultricies vehicula ut id elit. Cras mattis consectetur purus sit amet fermentum. Morbi leo risus, porta ac consectetur ac, vestibulum at eros.',
    button: 'Discover',
    image:
      'https://firebasestorage.googleapis.com/v0/b/bakery-dev-server.appspot.com/o/images%2Fmango_cheesecake.jpg?alt=media&token=b2ca7c3f-0197-435b-b17b-4687df8a8fbd',
    user: 'Samantha',
    userProfile: 'https://s7.postimg.cc/ujy8zz7vv/5_3x.png',
  },
  {
    title: 'Phasellus volutpat metus',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras mattis consectetur purus sit amet fermentum. Cras justo odio, dapibus ac facilisis in, egestas eget quam.',
    button: 'Buy now',
    image:
      'https://firebasestorage.googleapis.com/v0/b/bakery-dev-server.appspot.com/o/images%2Fbrownie_cake.jpg?alt=media&token=68896693-8bf8-4655-a4f3-88992ea32b15',
    user: 'Michael',
    userProfile: 'https://s7.postimg.cc/6exjimijv/3_3x.png',
  },
  {
    title: 'Ultricies Vulputate Mollis',
    description:
      'Aenean eu leo quam. Pellentesque ornare sem lacinia  nibh, ut fermentum massa justo sit amet risus. Cras justo odio, dapibus ac facilisis.',
    button: 'Read More',
    image:
      'https://firebasestorage.googleapis.com/v0/b/bakery-dev-server.appspot.com/o/images%2Ftart.jpg?alt=media&token=a0943076-4c69-4285-b432-38f6161d65b6',
    user: 'Jessica',
    userProfile: 'https://s7.postimg.cc/7ihnu80ij/4_3x.png',
  },
  {
    title: 'Odo Aenean Quam Tortor',
    description:
      'Nullam id dolor id nibh ultricies vehicula ut id elit. Cras mattis consectetur Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    button: 'Discover',
    image:
      'https://firebasestorage.googleapis.com/v0/b/bakery-dev-server.appspot.com/o/images%2Fbrownie.jpg?alt=media&token=6bd8070f-69f9-48fb-a3e8-008f78acbc7a',
    user: 'William',
    userProfile: 'https://s7.postimg.cc/f9ydt4zmj/2_3x.png',
  },
  {
    title: 'Volutpat Aenean metus',
    description:
      'quam venenatis vestibulum. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentumconsectetur purus sit amet fermentum. Cras justo odio, dapibus ac facilisis.',
    button: 'Buy now',
    image:
      'https://firebasestorage.googleapis.com/v0/b/bakery-dev-server.appspot.com/o/images%2Fcookieplate.jpg?alt=media&token=87eda7be-c9d5-4b0a-a773-469961b6da12',
    user: 'Katerina',
    userProfile: 'https://s7.postimg.cc/hsk2th5tn/6_3x.png',
  },
];
