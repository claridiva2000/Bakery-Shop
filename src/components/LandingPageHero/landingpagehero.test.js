import React from "react";
import renderer from "react-test-renderer";
import LandingPageHero from "./LandingPageHero";

it("should render LandingHero component", () => {
  const component = renderer.create(<LandingPageHero />).toJSON();
  expect(component).toMatchSnapshot();
  //this is a test showing that the button works
  //Assuming this is the order now custom button?
  //Test not passing
});
