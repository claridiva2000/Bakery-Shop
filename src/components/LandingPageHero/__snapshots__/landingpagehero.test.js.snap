// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LandingHero component 1`] = `
<div>
  <div
    className="slider"
  >
    <a
      className="previousButton"
      onClick={[Function]}
    >
      <svg
        height="30"
        viewBox="0 0 20 30"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          fill="#000"
          points="20 15 4.228 0 0 3.626 11.954 15 0 26.374 4.228 30"
          transform="rotate(180 10 15)"
        />
      </svg>
    </a>
    <a
      className="nextButton"
      onClick={[Function]}
    >
      <svg
        height="30"
        viewBox="0 0 20 30"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          fill="#000"
          points="20 15 4.228 0 0 3.626 11.954 15 0 26.374 4.228 30"
          transform="rotate(0 10 15)"
        />
      </svg>
    </a>
    <div
      className="track"
    >
      <div
        className="slide current slideBG"
        style={
          Object {
            "background": "url('https://firebasestorage.googleapis.com/v0/b/bakery-dev-server.appspot.com/o/images%2Fbanana%20pudding.png?alt=media&token=319d993f-6ebf-435d-bbd0-ca2c5e96d130') no-repeat center center",
            "backgroundSize": "cover",
            "outline": "200px solid rgba(0, 0, 0, 0.4)",
            "outlineOffset": "-200px",
            "overflow": "hidden",
          }
        }
      >
        <div
          className="center"
        >
          <h1>
            Vulputate Mollis Ultricies
          </h1>
          <p>
            Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis vestibulum. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh.
          </p>
          <button
            className="custom-button "
          >
            Order Now
          </button>
        </div>
      </div>
      <div
        className="slide next slideBG"
        style={
          Object {
            "background": "url('https://firebasestorage.googleapis.com/v0/b/bakery-dev-server.appspot.com/o/images%2Fmango_cheesecake.jpg?alt=media&token=b2ca7c3f-0197-435b-b17b-4687df8a8fbd') no-repeat center center",
            "backgroundSize": "cover",
            "outline": "200px solid rgba(0, 0, 0, 0.4)",
            "outlineOffset": "-200px",
            "overflow": "hidden",
          }
        }
      >
        <div
          className="center"
        >
          <h1>
            Tortor Dapibus
          </h1>
          <p>
            Nullam id dolor id nibh ultricies vehicula ut id elit. Cras mattis consectetur purus sit amet fermentum. Morbi leo risus, porta ac consectetur ac, vestibulum at eros.
          </p>
          <button
            className="custom-button "
          >
            Order Now
          </button>
        </div>
      </div>
      <div
        className="slide hidden slideBG"
        style={
          Object {
            "background": "url('https://firebasestorage.googleapis.com/v0/b/bakery-dev-server.appspot.com/o/images%2Fbrownie_cake.jpg?alt=media&token=68896693-8bf8-4655-a4f3-88992ea32b15') no-repeat center center",
            "backgroundSize": "cover",
            "outline": "200px solid rgba(0, 0, 0, 0.4)",
            "outlineOffset": "-200px",
            "overflow": "hidden",
          }
        }
      >
        <div
          className="center"
        >
          <h1>
            Phasellus volutpat metus
          </h1>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras mattis consectetur purus sit amet fermentum. Cras justo odio, dapibus ac facilisis in, egestas eget quam.
          </p>
          <button
            className="custom-button "
          >
            Order Now
          </button>
        </div>
      </div>
      <div
        className="slide hidden slideBG"
        style={
          Object {
            "background": "url('https://firebasestorage.googleapis.com/v0/b/bakery-dev-server.appspot.com/o/images%2Ftart.jpg?alt=media&token=a0943076-4c69-4285-b432-38f6161d65b6') no-repeat center center",
            "backgroundSize": "cover",
            "outline": "200px solid rgba(0, 0, 0, 0.4)",
            "outlineOffset": "-200px",
            "overflow": "hidden",
          }
        }
      >
        <div
          className="center"
        >
          <h1>
            Ultricies Vulputate Mollis
          </h1>
          <p>
            Aenean eu leo quam. Pellentesque ornare sem lacinia  nibh, ut fermentum massa justo sit amet risus. Cras justo odio, dapibus ac facilisis.
          </p>
          <button
            className="custom-button "
          >
            Order Now
          </button>
        </div>
      </div>
      <div
        className="slide hidden slideBG"
        style={
          Object {
            "background": "url('https://firebasestorage.googleapis.com/v0/b/bakery-dev-server.appspot.com/o/images%2Fbrownie.jpg?alt=media&token=6bd8070f-69f9-48fb-a3e8-008f78acbc7a') no-repeat center center",
            "backgroundSize": "cover",
            "outline": "200px solid rgba(0, 0, 0, 0.4)",
            "outlineOffset": "-200px",
            "overflow": "hidden",
          }
        }
      >
        <div
          className="center"
        >
          <h1>
            Odo Aenean Quam Tortor
          </h1>
          <p>
            Nullam id dolor id nibh ultricies vehicula ut id elit. Cras mattis consectetur Lorem ipsum dolor sit amet, consectetur adipiscing elit.
          </p>
          <button
            className="custom-button "
          >
            Order Now
          </button>
        </div>
      </div>
      <div
        className="slide previous slideBG"
        style={
          Object {
            "background": "url('https://firebasestorage.googleapis.com/v0/b/bakery-dev-server.appspot.com/o/images%2Fcookieplate.jpg?alt=media&token=87eda7be-c9d5-4b0a-a773-469961b6da12') no-repeat center center",
            "backgroundSize": "cover",
            "outline": "200px solid rgba(0, 0, 0, 0.4)",
            "outlineOffset": "-200px",
            "overflow": "hidden",
          }
        }
      >
        <div
          className="center"
        >
          <h1>
            Volutpat Aenean metus
          </h1>
          <p>
            quam venenatis vestibulum. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentumconsectetur purus sit amet fermentum. Cras justo odio, dapibus ac facilisis.
          </p>
          <button
            className="custom-button "
          >
            Order Now
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
